fe/fe-core/src/main/java/org/apache/doris/datasource/iceberg/IcebergExternalTable.java

```diff
@@ -90,4 +199,318 @@ public long fetchRowCount() {
+    }
+
+    @Override
+    public Map<String, PartitionItem> getAndCopyPartitionItems(Optional<MvccSnapshot> snapshot) {
+        return Maps.newHashMap(getPartitionInfoFromCache().getNameToPartitionItem());
+    }
+
+    private IcebergPartitionInfo getPartitionInfoFromCache() {
+        makeSureInitialized();
+        Optional<SchemaCacheValue> schemaCacheValue = getSchemaCacheValue();
+        if (!schemaCacheValue.isPresent()) {
+            return new IcebergPartitionInfo();
+        }
+        return ((IcebergSchemaCacheValue) schemaCacheValue.get()).getPartitionInfo();
+    }
+
+    @Override
+    public PartitionType getPartitionType(Optional<MvccSnapshot> snapshot) {
+        makeSureInitialized();
+        return isSupportedRelatedTable() ? PartitionType.RANGE : PartitionType.UNPARTITIONED;
+    }
+
+    @Override
+    public Set<String> getPartitionColumnNames(Optional<MvccSnapshot> snapshot) throws DdlException {
+        return getPartitionColumnsFromCache().stream().map(Column::getName).collect(Collectors.toSet());
+    }
+
+    @Override
+    public List<Column> getPartitionColumns(Optional<MvccSnapshot> snapshot) {
+        return getPartitionColumnsFromCache();
+    }
+
+    private List<Column> getPartitionColumnsFromCache() {
+        makeSureInitialized();
+        Optional<SchemaCacheValue> schemaCacheValue = getSchemaCacheValue();
+        return schemaCacheValue
+                .map(cacheValue -> ((IcebergSchemaCacheValue) cacheValue).getPartitionColumns())
+                .orElseGet(Lists::newArrayList);
+    }
+
+    @Override
+    public MTMVSnapshotIf getPartitionSnapshot(String partitionName, MTMVRefreshContext context,
+                                               Optional<MvccSnapshot> snapshot) throws AnalysisException {
+        long latestSnapshotId = getPartitionInfoFromCache().getLatestSnapshotId(partitionName);
+        if (latestSnapshotId <= 0) {
+            throw new AnalysisException("can not find partition: " + partitionName);
+        }
+        return new MTMVVersionSnapshot(latestSnapshotId);
+    }
+
+    @Override
+    public MTMVSnapshotIf getTableSnapshot(MTMVRefreshContext context, Optional<MvccSnapshot> snapshot)
+            throws AnalysisException {
+        return new MTMVVersionSnapshot(getLatestSnapshotIdFromCache());
+    }
+
+    public long getLatestSnapshotIdFromCache() throws AnalysisException {
+        makeSureInitialized();
+        Optional<SchemaCacheValue> schemaCacheValue = getSchemaCacheValue();
+        if (!schemaCacheValue.isPresent()) {
+            throw new AnalysisException("Can't find schema cache of table " + name);
+        }
+        return ((IcebergSchemaCacheValue) schemaCacheValue.get()).getSnapshotId();
+    }
+
+    @Override
+    public boolean isPartitionColumnAllowNull() {
+        if (partitionColumns == null || partitionColumns.size() != 1) {
+            return false;
+        }
+        return partitionColumns.get(0).isAllowNull();
```

Comment:
Just return true directly?

Reply:
And the case needs to ensure that Iceberg writes null values, and the materialized view can refresh to null data normally